packages/dcos-integration-test/extra/test_service_discovery.py

```diff
@@ -125,6 +128,206 @@ def _pool_for_mesos_dns():
+DNSAddresses = collections.namedtuple("DNSAddresses", ["container", "agent", "auto"])
+MarathonAddresses = collections.namedtuple("MarathonAddresses", ["host", "container"])
+
+
+def get_ipv4_addresses(hostname):
+    res = socket.getaddrinfo(hostname, 0, family=socket.AF_INET, type=socket.SOCK_STREAM)
+    return frozenset([sockaddr[0] for (family, type, proto, canonname, sockaddr) in res])
+
+
+def get_dns_addresses_by_app_name(app_name):
+    container_ip_name = '{}.marathon.containerip.dcos.thisdcos.directory'.format(app_name)
+    agent_ip_name = '{}.marathon.agentip.dcos.thisdcos.directory'.format(app_name)
+    auto_ip_name = '{}.marathon.autoip.dcos.thisdcos.directory'.format(app_name)
+    container_ips = get_ipv4_addresses(container_ip_name)
+    agent_ips = get_ipv4_addresses(agent_ip_name)
+    auto_ips = get_ipv4_addresses(auto_ip_name)
+    return DNSAddresses(container_ips, agent_ips, auto_ips)
+
+
+def get_marathon_addresses_by_service_points(service_points):
+    marathon_host_addrs = frozenset([point.host for point in service_points])
+    marathon_ip_addrs = frozenset([point.ip for point in service_points])
+    return MarathonAddresses(marathon_host_addrs, marathon_ip_addrs)
+
+
+def test_if_service_discovery_works_navstar_mesos_container_host_network(dcos_api_session):
+    app_definition, test_uuid = get_test_app_in_ucr()
+    with dcos_api_session.marathon.deploy_and_cleanup(app_definition) as service_points:
+        marathon_addrs = get_marathon_addresses_by_service_points(service_points)
+        assert marathon_addrs.host == marathon_addrs.container
+
+        @retrying.retry(wait_fixed=1000,
+                        stop_max_delay=DNS_ENTRY_UPDATE_TIMEOUT * 1000,
+                        retry_on_exception=lambda x: True)
+        def _ensure_dns_converged():
+            app_name = app_definition['id']
+            dns_addrs = get_dns_addresses_by_app_name(app_name)
+            assert marathon_addrs.host == dns_addrs.agent
+            assert marathon_addrs.host == dns_addrs.auto
+            assert marathon_addrs.host == dns_addrs.container
+
+        _ensure_dns_converged()
+
+
+def test_if_service_discovery_works_navstar_mesos_container_overlay_network(dcos_api_session):
+    port = 9080
+    app_definition, test_uuid = get_test_app_in_ucr()
+
+    app_definition['ipAddress'] = {
+        'networkName': 'dcos',
+        'discovery': {
+            'ports': [{
+                'protocol': 'tcp',
+                'name': 'test',
+                'number': port,
+            }]
+        }
+    }
+    if 'portIndex' in app_definition['healthChecks'][0]:
+        del app_definition['healthChecks'][0]['portIndex']
+    app_definition['healthChecks'][0]['port'] = port
+
+    # Take current definition, strip of the last word "$PORT0" because we
+    # aren't using the port index syntax, and then add on the port we're
+    # actually using.
+    cmd_list = app_definition['cmd'].split()[:-1]
```

Comment:
would it make sense to refactor this in a separate function? Looks like these steps are repeated in other tests as well